rfcs/convergence/authoring-stories.md

```diff
@@ -267,13 +270,40 @@ TBA
 
 TBA
 
-### 7. how structure stories in the storybook nav tree
+### 7. dissecting big story files into smaller ones
 
-TBA
+Components with complex API require many long stories, both for documentation and for e2e testing. This creates long files with hundres of lines, which are hard to maintain. How can these files be dissected into smaller ones?
+
+**Proposal:**
 
-### 8. dissecting big story files into smaller ones
+1. Every component can have at most one `.stories.tsx` file. This file has a default export with Storybook configuration for the docs page.
```

Comment:
Do you have any vision to enforcing this requirement ? It should be quite simply to use multiple `stories.tsx` files and still not change the storybook structurally at all

Reply:
it's why I kinda liked the `stories/` folder option I originally proposed, since we can force storybook to load folders from a specific folder instead of all wildcard blobs

Reply:
🤔 We could contact storybook maintainers and see if they’d be open to a PR which would enforce unique `title`s throughout `.stories.tsx` files, in an opt-in mode. I believe they could be open to it as they state in their [documentation](https://storybook.js.org/docs/react/api/csf) that titles should be unique. If titles are unique, duplicate `.stories.tsx` would be always shown as different pages on the doc site, so this would be easier to keep in check.

That’d be the vision. I am not sure if it’ll actually get done :/.

> it's why I kinda liked the stories/ folder option I originally proposed, since we can force storybook to load folders from a specific folder instead of all wildcard blobs

Putting all stories into one folder doesn’t solve the issue though, right? In one folder, you could still have multiple `.stories.tsx` files for one component.

Reply:
> Do you have any vision to enforcing this requirement ?

we can/want enforce guides via generators + lint/additional checks will need to be implemented manually...

but generators is primary point of HOW TO, as it if we decide to change things in future, based on that generator we can prepare migration generators and automate stuff